From 16850713c8953ea036edddb323b039cb8a731fa1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Thu, 22 Oct 2026 13:57:28 +0200
Subject: [PATCH] Fixes for the panel applet in in-process mode

Use glib/gi18n-lib.h instead of glib/gi18n.h to ensure that translations
use proper gettext domain.

Also, don't use gtk_show_about_dialog(), because this will interfere with
other applets.
---
 plugins/acpi/acpi-plugin.c            | 2 +-
 plugins/i2c-proc/i2c-proc-plugin.c    | 2 +-
 plugins/i2c-sys/i2c-sys-plugin.c      | 2 +-
 plugins/i8k/i8k-plugin.c              | 2 +-
 plugins/ibm-acpi/ibm-acpi-plugin.c    | 2 +-
 plugins/nvidia/nvidia-plugin.c        | 2 +-
 plugins/omnibook/omnibook-plugin.c    | 2 +-
 plugins/pmu-sys/pmu-sys-plugin.c      | 2 +-
 plugins/smu-sys/smu-sys-plugin.c      | 2 +-
 plugins/sonypi/sonypi-plugin.c        | 2 +-
 sensors-applet/about-dialog.c         | 7 +++++--
 sensors-applet/active-sensor.c        | 2 +-
 sensors-applet/prefs-dialog.c         | 2 +-
 sensors-applet/sensor-config-dialog.c | 2 +-
 sensors-applet/sensors-applet.c       | 2 +-
 15 files changed, 19 insertions(+), 16 deletions(-)

diff --git a/plugins/acpi/acpi-plugin.c b/plugins/acpi/acpi-plugin.c
index fc361139..72bdc35d 100644
--- a/plugins/acpi/acpi-plugin.c
+++ b/plugins/acpi/acpi-plugin.c
@@ -23,7 +23,7 @@
 
 #include <stdio.h>
 #include <glib.h>
-#include <glib/gi18n.h>
+#include <glib/gi18n-lib.h>
 #include "acpi-plugin.h"
 
 const gchar *plugin_name = "acpi";
diff --git a/plugins/i2c-proc/i2c-proc-plugin.c b/plugins/i2c-proc/i2c-proc-plugin.c
index 4a63a8ac..c26df6a1 100644
--- a/plugins/i2c-proc/i2c-proc-plugin.c
+++ b/plugins/i2c-proc/i2c-proc-plugin.c
@@ -24,7 +24,7 @@
 #include <stdio.h>
 #include <locale.h>
 #include <glib.h>
-#include <glib/gi18n.h>
+#include <glib/gi18n-lib.h>
 #include "i2c-proc-plugin.h"
 
 const gchar *plugin_name = "i2c-proc";
diff --git a/plugins/i2c-sys/i2c-sys-plugin.c b/plugins/i2c-sys/i2c-sys-plugin.c
index 411f98de..fea0990d 100644
--- a/plugins/i2c-sys/i2c-sys-plugin.c
+++ b/plugins/i2c-sys/i2c-sys-plugin.c
@@ -23,7 +23,7 @@
 
 #include <stdio.h>
 #include <glib.h>
-#include <glib/gi18n.h>
+#include <glib/gi18n-lib.h>
 #include "i2c-sys-plugin.h"
 
 const gchar *plugin_name = "i2c-sys";
diff --git a/plugins/i8k/i8k-plugin.c b/plugins/i8k/i8k-plugin.c
index e213d045..47d47309 100644
--- a/plugins/i8k/i8k-plugin.c
+++ b/plugins/i8k/i8k-plugin.c
@@ -23,7 +23,7 @@
 
 #include <stdio.h>
 #include <glib.h>
-#include <glib/gi18n.h>
+#include <glib/gi18n-lib.h>
 #include "i8k-plugin.h"
 
 const gchar *plugin_name = "i8k";
diff --git a/plugins/ibm-acpi/ibm-acpi-plugin.c b/plugins/ibm-acpi/ibm-acpi-plugin.c
index ae8f8ab9..803e1260 100644
--- a/plugins/ibm-acpi/ibm-acpi-plugin.c
+++ b/plugins/ibm-acpi/ibm-acpi-plugin.c
@@ -23,7 +23,7 @@
 
 #include <stdio.h>
 #include <glib.h>
-#include <glib/gi18n.h>
+#include <glib/gi18n-lib.h>
 #include "ibm-acpi-plugin.h"
 
 const gchar *plugin_name = "ibm-acpi";
diff --git a/plugins/nvidia/nvidia-plugin.c b/plugins/nvidia/nvidia-plugin.c
index 2643a2a6..62c620c7 100644
--- a/plugins/nvidia/nvidia-plugin.c
+++ b/plugins/nvidia/nvidia-plugin.c
@@ -27,7 +27,7 @@
 #include <NVCtrl/NVCtrlLib.h>
 
 #include <glib.h>
-#include <glib/gi18n.h>
+#include <glib/gi18n-lib.h>
 #include "nvidia-plugin.h"
 
 const gchar *plugin_name = "nvidia";
diff --git a/plugins/omnibook/omnibook-plugin.c b/plugins/omnibook/omnibook-plugin.c
index 7728110a..fe24b2fd 100644
--- a/plugins/omnibook/omnibook-plugin.c
+++ b/plugins/omnibook/omnibook-plugin.c
@@ -23,7 +23,7 @@
 
 #include <stdio.h>
 #include <glib.h>
-#include <glib/gi18n.h>
+#include <glib/gi18n-lib.h>
 #include "omnibook-plugin.h"
 
 const gchar *plugin_name = "omnibook";
diff --git a/plugins/pmu-sys/pmu-sys-plugin.c b/plugins/pmu-sys/pmu-sys-plugin.c
index 6b507423..e200a501 100644
--- a/plugins/pmu-sys/pmu-sys-plugin.c
+++ b/plugins/pmu-sys/pmu-sys-plugin.c
@@ -23,7 +23,7 @@
 
 #include <stdio.h>
 #include <glib.h>
-#include <glib/gi18n.h>
+#include <glib/gi18n-lib.h>
 #include "pmu-sys-plugin.h"
 
 const gchar *plugin_name = "pmu-sys";
diff --git a/plugins/smu-sys/smu-sys-plugin.c b/plugins/smu-sys/smu-sys-plugin.c
index f2c93eaa..961fbd29 100644
--- a/plugins/smu-sys/smu-sys-plugin.c
+++ b/plugins/smu-sys/smu-sys-plugin.c
@@ -23,7 +23,7 @@
 
 #include <stdio.h>
 #include <glib.h>
-#include <glib/gi18n.h>
+#include <glib/gi18n-lib.h>
 #include "smu-sys-plugin.h"
 
 const gchar *plugin_name = "smu-sys";
diff --git a/plugins/sonypi/sonypi-plugin.c b/plugins/sonypi/sonypi-plugin.c
index 64a0c8c1..2d21d877 100644
--- a/plugins/sonypi/sonypi-plugin.c
+++ b/plugins/sonypi/sonypi-plugin.c
@@ -34,7 +34,7 @@
 #endif /* HAVE_UNISTD_H */
 
 #include <glib.h>
-#include <glib/gi18n.h>
+#include <glib/gi18n-lib.h>
 #include "sonypi-plugin.h"
 
 #include <stdint.h>
diff --git a/sensors-applet/about-dialog.c b/sensors-applet/about-dialog.c
index 86465ce1..b6d133b9 100644
--- a/sensors-applet/about-dialog.c
+++ b/sensors-applet/about-dialog.c
@@ -22,7 +22,7 @@
 #endif /* HAVE_CONFIG_H */
 
 #include <glib.h>
-#include <glib/gi18n.h>
+#include <glib/gi18n-lib.h>
 #include <gtk/gtk.h>
 #include "about-dialog.h"
 
@@ -34,7 +34,7 @@ void about_dialog_open(SensorsApplet *sensors_applet) {
     };
 
     /* Construct the about dialog */
-    gtk_show_about_dialog(NULL,
+    GtkWidget *about = g_object_new (GTK_TYPE_ABOUT_DIALOG,
                   "icon-name", "mate-sensors-applet",
                   "program-name", PACKAGE_NAME,
                   "version", PACKAGE_VERSION,
@@ -49,4 +49,7 @@ void about_dialog_open(SensorsApplet *sensors_applet) {
                   "website", PACKAGE_URL,
                   NULL);
 
+    g_signal_connect_swapped (about, "response",
+                              G_CALLBACK (gtk_widget_destroy), about);
+    gtk_widget_show (about);
 }
diff --git a/sensors-applet/active-sensor.c b/sensors-applet/active-sensor.c
index fdd10c30..0fae6f47 100644
--- a/sensors-applet/active-sensor.c
+++ b/sensors-applet/active-sensor.c
@@ -26,7 +26,7 @@
 #endif /* HAVE_STRING_H */
 
 #include <glib.h>
-#include <glib/gi18n.h>
+#include <glib/gi18n-lib.h>
 #include <gtk/gtk.h>
 #include <gio/gio.h>
 
diff --git a/sensors-applet/prefs-dialog.c b/sensors-applet/prefs-dialog.c
index 20db2ec2..4b02ecf0 100644
--- a/sensors-applet/prefs-dialog.c
+++ b/sensors-applet/prefs-dialog.c
@@ -21,7 +21,7 @@
 #include "config.h"
 #endif /* HAVE_CONFIG_H */
 
-#include <glib/gi18n.h>
+#include <glib/gi18n-lib.h>
 #include <gio/gio.h>
 #include "sensors-applet-settings.h"
 #include "prefs-dialog.h"
diff --git a/sensors-applet/sensor-config-dialog.c b/sensors-applet/sensor-config-dialog.c
index 7a85d1b7..68cd55d9 100644
--- a/sensors-applet/sensor-config-dialog.c
+++ b/sensors-applet/sensor-config-dialog.c
@@ -21,7 +21,7 @@
 #include "config.h"
 #endif /* HAVE_CONFIG_H */
 
-#include <glib/gi18n.h>
+#include <glib/gi18n-lib.h>
 #include "sensor-config-dialog.h"
 #include "sensors-applet.h"
 
diff --git a/sensors-applet/sensors-applet.c b/sensors-applet/sensors-applet.c
index 93c8617f..4e9d7e54 100644
--- a/sensors-applet/sensors-applet.c
+++ b/sensors-applet/sensors-applet.c
@@ -29,7 +29,7 @@
 #include <unistd.h>
 #endif /* HAVE_UNISTD_H */
 
-#include <glib/gi18n.h>
+#include <glib/gi18n-lib.h>
 #include <glib/gprintf.h>
 #include <gio/gio.h>
 #include "sensors-applet.h"
